lib/switchtec.c

```diff
@@ -212,7 +212,8 @@ struct switchtec_dev *switchtec_open(const char *device)
 	char *endptr;
 	struct switchtec_dev *ret;
 
-	if (sscanf(device, "%2049[^:]:%i", path, &inst) == 2) {
+	if (sscanf(device, "%2049[^:]:%i%n", path, &inst, &idx) == 2 &&
+	    device[idx]=='\0') {
```

Comment:
It's too tricky I guess. Can we use a format like what we do for pci address at line 242?

Reply:
Then we need to change the `switchtec_open_eth(path, inst)` function format. Do you want to do this?

Reply:
Oh, yeah, this doesn't look spectacular... Why are we trying to match one of the most general methods first? It looks like it also breaks open_i2c with a colon below as well.

We should either try this last and allow any hostname  even without the instance number (which looks oddly conflated with a port number) or we should make the ethernet option require a better indication that the user really means to connect via ethernet, like using a URL. This seems like the best choice because the user should really be able to optionally specify the port. 

"switchtec://hostname[:port]/[inst]" would make a bit more sense for ethernet.